energy/energy_generation_consumption.py

```diff
@@ -4,76 +4,114 @@
 
 # COMMAND ----------
 
-# Source prevents programmatic data fetch: "https://www.energyinst.org/__data/assets/file/0015/1421601/2023-Country-Transition-Tracker-Consolidated-Format.csv"
-current_dir = Path().resolve()
-file_path = current_dir / '2023-Country-Transition-Tracker-Consolidated-Format.csv'
+# Source prevents programmatic data fetch: https://ember-climate.org/data/api/
+# Curretnly requesting ITS to whitelist this request
+
+file_path = "/Workspace/Users/ysuzuki2@worldbank.org/mega-indicators/energy/yearly_full_release_long_format.csv"
 raw_df = pd.read_csv(file_path)
-raw_df
+country_df = (
+    spark.table(f"indicator.country")
+    .select("country_name", "country_code", "region")
+    .toPandas()
+)
+energy_df = raw_df.merge(country_df, left_on="Country code", right_on="country_code")
 
 # COMMAND ----------
 
-raw_df = raw_df.rename(columns={"Country": "country_name", "Year": "year"})
-
-# Normalizing to WB economy names
-name_map = {
-    'China Hong Kong SAR': 'Hong Kong SAR, China',
-    "Czech Republic": "Czechia",
-    "Egypt": "Egypt, Arab Rep.",
-    "Iran": "Iran, Islamic Rep.",
-    "Slovakia": "Slovak Republic",
-    "South Korea": "Korea, Rep.",
-    "Trinidad & Tobago": "Trinidad and Tobago",
-    "Turkey": "Turkiye",
-    "US": "United States",
-    "Venezuela": "Venezuela, RB",
-    "Vietnam": "Viet Nam",
-}
-for old_name, new_name in name_map.items():
-    raw_df.loc[raw_df.country_name == old_name, 'country_name'] = new_name
+# These areas were excluded by the merge. But none of them were recognized country
 
-# COMMAND ----------
+emmited_areas = [
+    country
+    for country in raw_df["Area"].unique()
+    if country not in energy_df["Area"].unique()
+]
+emmited_areas
 
-df_renewable = raw_df[raw_df.Variable == 'Primary Renewable Energy Consumption']
-df_renewable = df_renewable.drop(columns=["Unit", "Variable"])\
-    .rename(columns={"Value": "primary_renewable_consumption_share"})
-
-df_total = raw_df[raw_df.Variable == 'Primary Energy Consumption per Capita']
-df_total = df_total.drop(columns=["Unit", "Variable"])\
-    .rename(columns={"Value": "primary_consumption_gigajoules_per_capita"})
+# COMMAND ----------
 
-df = pd.merge(df_renewable, df_total, on=['country_name', 'year'], how='left')
-df
+# Define the fuel types based on the data methodology documentation https://ember-climate.org/data-catalogue/yearly-electricity-data/|
```

Comment:
These types are based on the data documentation 
![image](https://github.com/user-attachments/assets/236f4668-f46b-4094-a160-2d2c8d71a36e)